caas/kubernetes/provider/labels.go

```diff
@@ -3,44 +3,201 @@
+	// created by applications managed by Juju. Think istio, kubeflow etc
+	// See https://bugs.launchpad.net/juju/+bug/1892285
+	LabelJujuAppCreatedBy = "app.juju.is/created-by"
+
+	// LabelJujuModelName is the juju label applied for juju models.
+	LabelJujuModelName = "model.juju.is/name"
+
+	// LabelJujuOperatorName is the juju label applied to Juju operators to
+	// identify their name. Operator names are generally named after the thing
+	// the operator is controlling. i.e an operator name for a model test would be
+	// "test"
+	LabelJujuOperatorName = "operator.juju.is/name"
+
+	// LabelJujuOperatorTarget is the juju label applied to Juju operators to
+	// describe the modeling paradigm they target. For example model,
+	// application
+	LabelJujuOperatorTarget = "operator.juju.is/target"
+
+	// LabelJujuStorageName is the juju label applied to Juju storage objects to
+	// describe their name.
+	LabelJujuStorageName = "storage.juju.is/name"
+
+	// legacyLabelKubernetesAppName is the legacy label key used for juju app
+	// identification. This purely exists to maintain backwards functionality.
+	// See https://bugs.launchpad.net/juju/+bug/1888513
+	legacyLabelKubernetesAppName = "juju-app"
+
+	// legacyLabelModelName is the legacy label key used for juju models. This
+	// purely exists to maintain backwards functionality.
+	// See https://bugs.launchpad.net/juju/+bug/1888513
+	legacyLabelModelName = "juju-model"
+
+	// legacyLabelKubernetesOperatorName is the legacy label key used for juju
+	// operators. This purely exists to maintain backwards functionality.
+	// See https://bugs.launchpad.net/juju/+bug/1888513
+	legacyLabelKubernetesOperatorName = "juju-operator"
+
+	// legacyLabelJujuStorageName is the legacy label key used for juju storage
+	// pvc. This purely exists to maintain backwards functionality.
+	// See https://bugs.launchpad.net/juju/+bug/1888513
+	legacyLabelStorageName = "juju-storage"
+)
+
+var (
+	// LabelsJuju is a common set
+	LabelsJuju = map[string]string{
+		LabelKubernetesAppManaged: "juju",
 	}
-	return dest
+)
+
+func AppLabelSelector(appName string) (labels.Selector, error) {
+	return labels.Parse(fmt.Sprintf("%s=%s", LabelKubernetesAppName, appName))
 }
 
 func (k *kubernetesClient) getlabelsForApp(appName string, isNamespaced bool) map[string]string {
-	labels := LabelsForApp(appName)
+	labels := LabelsForApp(appName, k.IsLegacyLabels())
 	if !isNamespaced {
-		labels = AppendLabels(labels, LabelsForModel(k.CurrentModel()))
+		labels = LabelsMerge(labels, LabelsForModel(k.CurrentModel(), k.IsLegacyLabels()))
 	}
 	return labels
 }
 
+// HasLabels returns true if the src contains the labels in has
+func HasLabels(src, has labels.Set) bool {
+	for k, v := range has {
+		if src[k] != v {
+			return false
+		}
+	}
+	return true
+}
+
+// LabelSetToSelector converts a set of Kubernetes labels
+func LabelSetToSelector(l labels.Set) labels.Selector {
```

Comment:
These don't need to be exported either
Well, LabelsForModel does, but a bunch of others no